### ISSUE TITLE
Time from donor

### ISSUE THREAD

**Issue by pbastide** [bomeara/BMhyb#13]
Hi @bomeara  and @djhwueng,

I have been trying to test your function `GetVModified` on the example network you show on your [preprint](https://www.biorxiv.org/content/early/2015/08/05/023986), but I ran into a problem. 

I used the following function to create this network (with `t1`, `t2` and `t3` as in the preprint):
```R
create_paper_network <- function(gamma, t1, t2, t3){
    phy <- read.tree(text = paste0("((R:", t3, ",Y:", t3, "):", t1 + t2, ",X:", t1 + t2 + t3, ");"))
    network <- list(phy = phy,
                    flow = data.frame(donor = "X",
                                      recipient = "R",
                                      gamma = gamma,
                                      time.from.root.donor = t1,
                                      time.from.root.recipient = t1 + t2))
    network$flow$donor <- as.character(network$flow$donor)
    network$flow$recipient <- as.character(network$flow$recipient)
    return(network)
}
```
To plot an example:
```R
gamma <- 0.5
t1 <- 0.3; t2 <- 0.4; t3 <- 0.3; # unit height
network <- create_paper_network(gamma, t1, t2, t3)

PlotNetwork(network$phy, network$flow)
axis(1, at = c(0, t1, t1+t2, t1+t2+t3), labels = c("0", "t1", "t1+t2", "t1+t2+t3"))
```
Is this network correct ? I tried to copy the format given by outputs of your function `SimulateNetwork`, but I might have made a mistake.

Using this network, I had a problem computing the induced variance matrix using `GetVModified`:
```R
sigma2 = 1
x <- c(sigma.sq = sigma2, mu = 0, SE = 0)
actual.params <- c("sigma.sq", "mu", "bt", "vh", "SE")

vcv_BMhyb <- GetVModified(x, network$phy, network$flow, actual.params)
```
This gave me the following result:
```R
     R   Y    X
R 0.65 0.7 0.35
Y 0.70 1.0 0.00
X 0.35 0.0 1.00
```
There is a problem here with Cov[Y,R] and Cov[X,R]. Applying the formulas, I get:
```tex
Cov[X, R] = sigma^2 * gamma * t1 = 0.5*0.3 = 0.15 \neq 0.35
Cov[Y, R] = sigma^2 * (1-gamma) * (t1 + t2) = 0.35 \neq 0.7
```
I could not explain this discrepancy. Did I misused your functions ? Or are my computations wrong ?

One point that is unclear to me, is that I could find no reference to the parameters `time.from.root.donor` (t1) in the code of `GetVModified`, that seems essential for the computation of this matrix (but maybe it's hidden in the call of an other function, in which case I might have missed it).

Thank you for your help, and for your package !

Session infos:
```R
> sessionInfo()
R version 3.4.2 (2017-09-28)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.3 LTS

Matrix products: default
BLAS: /usr/lib/openblas-base/libblas.so.3
LAPACK: /usr/lib/libopenblasp-r0.2.18.so

locale:
[1] LC_CTYPE=fr_FR.UTF-8       LC_NUMERIC=C               LC_TIME=fr_FR.UTF-8        LC_COLLATE=fr_FR.UTF-8    
[5] LC_MONETARY=fr_FR.UTF-8    LC_MESSAGES=fr_FR.UTF-8    LC_PAPER=fr_FR.UTF-8       LC_NAME=C                 
[9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=fr_FR.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] BMhyb_1.5.1 ape_4.1    

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.12            subplex_1.4-1           msm_1.6.4               mvtnorm_1.0-6          
 [5] lattice_0.20-35         tidyr_0.7.1             corpcor_1.6.9           prettyunits_1.0.2      
 [9] assertthat_0.2.0        digest_0.6.12           foreach_1.4.3           R6_2.2.2               
[13] plyr_1.8.4              phytools_0.6-20         coda_0.19-1             httr_1.3.1             
[17] ggplot2_2.2.1           progress_1.1.2          rlang_0.1.2.9000        uuid_0.1-2             
[21] lazyeval_0.2.0          curl_2.8.1              data.table_1.10.4       taxize_0.9.0           
[25] phangorn_2.2.0          Matrix_1.2-11           RNeXML_2.0.7            combinat_0.0-8         
[29] splines_3.4.2           stringr_1.2.0           igraph_1.1.2            munsell_0.4.3          
[33] compiler_3.4.2          numDeriv_2016.8-1       geiger_2.0.6            pkgconfig_2.0.1        
[37] mnormt_1.5-5            tibble_1.3.4            gridExtra_2.2.1         TreeSim_2.3            
[41] expm_0.999-2            quadprog_1.5-5          codetools_0.2-15        XML_3.98-1.9           
[45] reshape_0.8.7           viridisLite_0.2.0       dplyr_0.7.2             MASS_7.3-47            
[49] crul_0.3.8              grid_3.4.2              nlme_3.1-131            jsonlite_1.5           
[53] gtable_0.2.0            magrittr_1.5            scales_0.5.0            stringi_1.1.5          
[57] reshape2_1.4.2          viridis_0.4.0           bindrcpp_0.2            scatterplot3d_0.3-40   
[61] phylobase_0.8.4         xml2_1.1.1              fastmatch_1.1-0         deSolve_1.20           
[65] iterators_1.0.8         tools_3.4.2             rncl_0.8.2              ade4_1.7-8             
[69] bold_0.5.0              glue_1.1.1              purrr_0.2.3             maps_3.2.0             
[73] plotrix_3.6-6           parallel_3.4.2          survival_2.41-3         colorspace_1.3-2       
[77] bindr_0.1               animation_2.5           clusterGeneration_1.3.4
```

**Comment by bomeara** [comment]
Hi, @pbastide. The preprint is out of date -- we need to update it to match the SystBio submission. Most importantly, the formulas have been fixed (largely due to your input, in fact). In the submission, for example, the covariance formula is this: 
![screen shot 2017-10-10 at 10 24 20 am](https://user-images.githubusercontent.com/1612612/31391679-39c8f986-ada5-11e7-8ddd-63017b544400.jpg)

**Comment by pbastide** [comment]
Indeed, the version you give here is not the version I have. However, I disagree with the formulas you give here (assuming that they still apply to the same network than the one presented in the preprint?)

The formula you give here are:
```tex
Cov[X, R] = sigma^2 * gamma * (t1 + t2)
Cov[Y, R] = 2 * sigma^2 * (1 - gamma) * (t1 + t2)
```
If you take `gamma=0`, then you cancel the effect of the hybridization, and should get a simple tree, where R and X are not related anymore, and where the mrca of R and Y is in C, at time t1+t2. Hence, for `gamma=0`, you should get:
```tex
Cov[Y, R] = sigma^2 * (t1 + t2)
```
but, using your formula above, one get:
```tex
Cov[Y, R] = 2 * sigma^2 * (1 - 0) * (t1 + t2) = 2 * sigma^2 * (t1 + t2).
```

Similarly, If you take `gamma=1`, then you should get a simple tree, where R and Y are not related anymore, and where the mrca of R and X is in A, at time t1. Hence, for `gamma=1`, you should get:
```tex
Cov[X, R] = sigma^2 * t1
```
but, using your formula above, one gets:
```tex
Cov[X, R] = sigma^2 * (t1 + t2).
```

The comment above use the following formulas, which I think are correct:
```tex
Cov[X, R] = sigma^2 * gamma * t1
Cov[Y, R] = sigma^2 * (1 - gamma) * (t1 + t2).
```

**Comment by bomeara** [comment]
hmm. After posting my comment, I noticed that the paper had changed on github, so I'm not sure that the version I quoted is even the one we submitted. Let me coordinate more with Tony (I've been traveling) to figure this out.

**Comment by bomeara** [comment]
Yes, my apologies -- the version I quoted above has new formulas made in response to your email. I agree they're not right.

btw, thanks for the fast, frank feedback. This is a great example of the utility of peer review (with *good* peer reviewers).

**Comment by bomeara** [comment]
New ape (5) has networks -- this might be better than the implementation I've been trying to do

**Comment by bomeara** [comment]
Fixed using commit efbc205e5812d1b06c4d10133c4cc131211253b1

**Comment by pbastide** [comment]
Thank you @bomeara and @djhwueng for addressing this issue !

I'm not sure if you saw the comment I made earlier on the code (referenced above), so I'm tagging you here.

I agree that the value for `Cov[R, X]` is now correct, equal to `0.15`.

However, I don't understand why `Cov[R, Y]` should be equal to `0.7` ? That's the expected value in your test [here](https://github.com/bomeara/BMhyb/blob/efbc205e5812d1b06c4d10133c4cc131211253b1/tests/testthat/test_condition.R#L59), and, when I install the last version of `BMHyb` from master, that's also what I get.
But, according to the formula I wrote above, one should have:
```math
Cov[R, Y] = sigma^2 * (1-gamma) * (t1 + t2) = 1.0 * 0.5 * (0.3 + 0.4) = 0.35
```

Also, when I change the value of `gamma`, this covariance `Cov[R,Y]` does not change, so when `gamma=1`, it it still computed to be `0.7`, while it should be `0`, as the species are not related anymore in that limit tree case.

From your code, it's not clear when the covariance between a "recipient" (here, `R`) and a sibling that is not a "donor" (`Y`) gets updated, so it always seems to be equal to the tree covariance (`0.7`). 

But I did not go over the details of your implementation, so it's probable that I might be missing something. Or maybe I'm commenting too soon, and you are still working on it ? 

Thank you again for taking my comments into account, and for your answers !
